src/state_manager/mod.rs

```diff
@@ -1271,12 +1269,33 @@ where
             Some(child)
         });
 
+        self.validate_stream(epochs, tipsets)
+    }
+
+    pub fn validate_stream<F>(
+        self: &Arc<Self>,
+        epochs: RangeInclusive<i64>,
+        tipsets: itertools::Unfold<Option<Arc<Tipset>>, F>,
+    ) -> anyhow::Result<()>
+    where
+        F: FnMut(&mut Option<Arc<Tipset>>) -> Option<Arc<Tipset>> + Send,
+    {
+        use rayon::iter::ParallelIterator as _;
+
+        let pb = crate::utils::io::ProgressBar::new((epochs.end() - epochs.start()) as u64);
+        pb.message("Compute parent state: ");
+        pb.set_max_refresh_rate(Some(std::time::Duration::from_millis(500)));
```

Comment:
I'm not sold on this change. Let's keep things the way they were. If you feel this is an improvement, open an issue.

Reply:
sure